src/wallet/cqrs/create-wallet.ts

```diff
@@ -22,26 +24,49 @@ export class CreateWalletHandler
   ) {}
 
   async execute({ address }: CreateWalletCommand): Promise<WalletDto> {
-    const trimmedAddress = address.trim();
+    address = address.trim();
 
-    assertValidAddress(trimmedAddress);
-    await this.assertAddressDoesNotExist(trimmedAddress);
+    const validationData = await this.getExtraWalletValidationData(address);
 
-    const wallet = new Wallet(trimmedAddress);
+    this.assertCanCreate(address, validationData);
+
+    const wallet = new Wallet(address, validationData);
     await this.repository.save(wallet);
 
     return WalletDto.from(wallet);
   }
 
-  private async assertAddressDoesNotExist(address: string): Promise<void> {
-    const existingCount = await this.repository.count({
-      where: {
-        address: address.trim(),
+  private async getExtraWalletValidationData(
+    address: string,
+  ): Promise<ExtraWalletValidationData> {
+    const [existingCount] = await Promise.all([
```

Comment:
Why wrapping with `Promise.all`?

Reply:
Yes it's not necessary here, but If you would have several queries they would execute pretty fast inside Promise.all